quinn-proto/src/connection/mod.rs

```diff
@@ -2015,40 +2018,13 @@ impl Connection {
         ecn: Option<EcnCodepoint>,
         partial_decode: PartialDecode,
     ) {
-        let header_crypto = if partial_decode.is_0rtt() {
-            if let Some(ref crypto) = self.zero_rtt_crypto {
-                Some(&*crypto.header)
-            } else {
-                debug!("dropping unexpected 0-RTT packet");
-                return;
-            }
-        } else if let Some(space) = partial_decode.space() {
-            if let Some(ref crypto) = self.spaces[space].crypto {
-                Some(&*crypto.header.remote)
-            } else {
-                debug!(
-                    "discarding unexpected {:?} packet ({} bytes)",
-                    space,
-                    partial_decode.len(),
-                );
-                return;
-            }
-        } else {
-            // Unprotected packet
-            None
-        };
-
-        let packet = partial_decode.data();
-        let stateless_reset = packet.len() >= RESET_TOKEN_SIZE + 5
-            && self.peer_params.stateless_reset_token.as_deref()
-                == Some(&packet[packet.len() - RESET_TOKEN_SIZE..]);
-
-        match partial_decode.finish(header_crypto) {
-            Ok(packet) => self.handle_packet(now, remote, ecn, Some(packet), stateless_reset),
-            Err(_) if stateless_reset => self.handle_packet(now, remote, ecn, None, true),
-            Err(e) => {
-                trace!("unable to complete packet decoding: {}", e);
-            }
+        if let Some(decoded) = packet_crypto::unprotect_header(
```

Comment:
Oh, I really want this as one separate commit with just the extraction of this code into a separate method.